### PR TITLE
Update php packages

### PR DESCRIPTION
**Pull request by Izorkin**
###### Motivation for this change
Update php packages:
- php 7.2.21
- php 7.3.8
- composer 1.9.0
- event 2.5.3
- ast 1.0.3
- protobuf 3.9.0
- pthreads 3.2.0
- redis 5.0.2

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/top-level/php-packages.nix [hunk 1]
```diff
@@ -493,20 +493,41 @@ let
     };
   };
 
-  pthreads = assert (pkgs.config.php.zts or false); buildPecl rec {
-    version = "3.1.5";
+  pthreads = if isPhp73 then pthreads32-dev else pthreads32;
+
+  pthreads32 = assert (pkgs.config.php.zts or false); assert !isPhp73; buildPecl rec {
```
Comment (Izorkin) [inline pkgs/top-level/php-packages.nix]: `assert (pkgs.config.php.zts or false);`
I don’t know how to make it work correctly

**Comment by Izorkin** [comment]
cc @etu

File: pkgs/top-level/php-packages.nix (hunk 1 above)
Comment (etu) [reply pkgs/top-level/php-packages.nix]: What is the resulting behavior that you need?

**Review by etu — APPROVED** [review]
I've have ran through the whole nix-review. Everything did build on my system, except for the oci8 packages. But those failed due to missing source archives. And I'm not willing to get those because it requires an oracle account.

**Comment by etu** [comment]
The php upgrades should also be backported to 19.03

File: pkgs/top-level/php-packages.nix (hunk 1 above)
Comment (Izorkin) [reply pkgs/top-level/php-packages.nix]: If enable ztsSupport (config.php.zts or true) need build pthreads. If (pkgs.config.php.zts or false) - disable build

Comment (etu) [reply pkgs/top-level/php-packages.nix]: But not if it's not php73 I assume?

Comment (Izorkin) [reply pkgs/top-level/php-packages.nix]: yes

Comment (Izorkin) [reply pkgs/top-level/php-packages.nix]: to php 7.3 need pthreads 3.2.0-dev

**Comment by aanderse** [comment]
`oci8` builds against `php72` and `php73`. Don't hesitate to ping me if you ever need the `oci8` packages built for testing.

**Review by vcunat — APPROVED** [review]
Almost only patch updates, a couple positive reviews, and it's been a week already.

**Comment by vcunat** [comment]
I picked the PHP bumps to 19.03, but I haven't looked if anything else needs to.

**Pull request merged**